In [179]:
import pandas as pd
import numpy as np

In [180]:
categorical_columns = ['NU_ANO_CENSO', 'REGIAO', 'IN_CAPITAL', 'IN_ACESSO_PORTAL_CAPES', 'IN_REPOSITORIO_INSTITUCIONAL', 'IN_BUSCA_INTEGRADA', 'IN_SERVICO_INTERNET', 'IN_PARTICIPA_REDE_SOCIAL',
                       'IN_CATALOGO_ONLINE', 'IN_OUTRAS_BASES', 'NOME_CATEGORIA_ADMINISTRATIVA', 'NOME_ORGANIZACAO_ACADEMICA']
numeric_columns = ['QT_TEC_TOTAL', 'QT_PERIODICO_ELETRONICO', 'QT_LIVRO_ELETRONICO', 'QT_ALUNOS', 'QT_DOCENTES', 'QT_CURSO', 'T1_VL_RECEITAS', 'T1_VL_DESPESAS']

In [181]:
pd.set_option('display.float_format', '{:,.2f}'.format)

# Analises dos clusters

In [182]:
def get_total_categorical_values_cluster(df):
    for col in categorical_columns:
        items = df[col].value_counts().items()

        print(f"====== {col} ======")

        for item in items:
            print(f"{item[0]}: {item[1]} ({(item[1] / df.shape[0]): 0.2%})")

        print()


# Agglomerative

### First cluster

In [183]:
first_cluster_agg = pd.read_csv('./data/Agglomerative/Primeiro agrupamento (Agglomerative).csv', sep=';')

In [184]:
get_total_categorical_values_cluster(first_cluster_agg)

====== NU_ANO_CENSO ======
2016: 241 ( 25.42%)
2017: 239 ( 25.21%)
2018: 238 ( 25.11%)
2019: 230 ( 24.26%)

====== REGIAO ======
Sudeste: 368 ( 38.82%)
Sul: 213 ( 22.47%)
Nordeste: 198 ( 20.89%)
Norte: 95 ( 10.02%)
Centro-Oeste: 74 ( 7.81%)

====== IN_CAPITAL ======
0: 499 ( 52.64%)
1: 449 ( 47.36%)

====== IN_ACESSO_PORTAL_CAPES ======
1: 871 ( 91.88%)
0: 77 ( 8.12%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 580 ( 61.18%)
0: 368 ( 38.82%)

====== IN_BUSCA_INTEGRADA ======
1: 579 ( 61.08%)
0: 369 ( 38.92%)

====== IN_SERVICO_INTERNET ======
1: 910 ( 95.99%)
0: 38 ( 4.01%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 711 ( 75.00%)
0: 237 ( 25.00%)

====== IN_CATALOGO_ONLINE ======
1: 853 ( 89.98%)
0: 95 ( 10.02%)

====== IN_OUTRAS_BASES ======
1.0: 631 ( 66.56%)
0.0: 317 ( 33.44%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Pública Federal: 411 ( 43.35%)
Privada sem fins lucrativos: 298 ( 31.43%)
Pública Estadual: 152 ( 16.03%)
Privada com fins lucrativos: 65 ( 6.86%)
Pública Munici

In [185]:
first_cluster_agg[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count        948.00                   948.00               948.00     948.00   
mean       1,067.03                11,002.45            30,041.69  19,394.41   
std        1,358.21                36,247.98            90,346.03  21,687.59   
min           33.00                     0.00                 0.00   1,005.00   
25%          398.75                     0.00                 0.00   6,462.75   
50%          707.00                    31.00             2,655.00  12,731.00   
75%        1,189.75                 3,525.00            12,671.00  26,057.75   
max       15,562.00               572,574.00         1,000,000.00 245,102.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS    T1_VL_DESPESAS  
count       948.00    948.00           948.00            948.00  
mean        962.64     67.27   454,150,088.54    479,909,885.36  
std         836.70     48.33   653,881,575.29    852,086,400.18  
min          60.00      8.00       134,763.64     10,769,001.91  
25%         413.75     34.00   130,128,444.86    147,817,024.68  
50%         704.00     53.00   276,350,996.41    278,105,501.08  
75%       1,228.50     92.00   514,270,381.47    499,588,321.00  
max       6,286.00    321.00 7,599,192,871.00 15,890,130,144.40

Conclusões:

- Predominância das regiões Sudeste, Sul e Nordeste  (38.82%, 22.47%, 20.89%)
- Instituições com acesso ao portal Capes (91.88%)
- Predominancia de repositorio institucional (61.18%)
- Predominancia busca integrada (61.08%)
- Possuem serviço de internet (95.99%)
- Participam em redes sociais (75.00%)
- Possuem catalogo online (89.98%)
- Possuem outras bases (66.56%)
- IES publica federal, sem fins lucrativos (43.35%, 31.43%)
- Mais da metade das IES são do tipo universidade e Instituto Federal de Educação, Ciência e Tecnologia (75.32%, 16.03%)

- Quantidade alta de técnicos na média (1,067.03)
- Quantidade de alunos média de alunos alto (19,394.41)
- Corpo docente forte na média (962.64	)
- Quantidade considerável de cursos na média (média de 67.27)
- Despesas e receitas média alta
- Alta quantidade de Quantidade de periodicos e livros eletronicos
- Quantidade de periodicos e livros eletronicos com relevância alto

### Second cluster

In [186]:
second_cluster_agg = pd.read_csv('./data/Agglomerative/Segundo agrupamento (Agglomerative).csv', sep=';')

In [187]:
get_total_categorical_values_cluster(second_cluster_agg)

====== NU_ANO_CENSO ======
2016: 577 ( 27.03%)
2017: 558 ( 26.14%)
2019: 524 ( 24.54%)
2018: 476 ( 22.30%)

====== REGIAO ======
Sudeste: 833 ( 39.02%)
Nordeste: 551 ( 25.81%)
Sul: 396 ( 18.55%)
Centro-Oeste: 218 ( 10.21%)
Norte: 137 ( 6.42%)

====== IN_CAPITAL ======
0: 1393 ( 65.25%)
1: 742 ( 34.75%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 1670 ( 78.22%)
1: 465 ( 21.78%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
0: 1442 ( 67.54%)
1: 693 ( 32.46%)

====== IN_BUSCA_INTEGRADA ======
1: 1298 ( 60.80%)
0: 837 ( 39.20%)

====== IN_SERVICO_INTERNET ======
1: 1790 ( 83.84%)
0: 345 ( 16.16%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 1402 ( 65.67%)
0: 733 ( 34.33%)

====== IN_CATALOGO_ONLINE ======
1: 1119 ( 52.41%)
0: 1016 ( 47.59%)

====== IN_OUTRAS_BASES ======
0.0: 1323 ( 61.97%)
1.0: 812 ( 38.03%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada com fins lucrativos: 1285 ( 60.19%)
Privada sem fins lucrativos: 770 ( 36.07%)
Pública Municipal e Especial: 64 ( 3.00%)
Pública Estadual

In [188]:
second_cluster_agg[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      2,135.00                 2,135.00             2,135.00   2,135.00   
mean          14.10                   398.74             3,867.71     282.62   
std           18.39                 2,774.72           139,084.54     341.36   
min            1.00                     0.00                 0.00       1.00   
25%            6.00                     0.00                 0.00      68.00   
50%           10.00                     5.00                 0.00     166.00   
75%           16.00                    48.00                33.50     372.00   
max          449.00                54,365.00         6,424,945.00   3,633.00   

       QT_DOCENTES  QT_CURSO  T1_VL_RECEITAS  T1_VL_DESPESAS  
count     2,135.00  2,135.00        2,135.00        2,135.00  
mean         21.21      2.75    1,982,280.26    2,025,236.50  
std          17.92      2.34    4,747,147.30    6,470,177.04  
min           1.00      1.00            0.01            2.00  
25%          10.00      1.00      337,634.51      351,376.38  
50%          16.00      2.00      886,312.00      886,696.00  
75%          27.00      3.00    1,931,850.57    2,007,175.73  
max         280.00     24.00   95,894,788.00  191,892,519.06

Conclusões:

- Predominância das regiões Sudeste e Nordeste (39.02%, 25.81% )
- Uma incidência das IES não estarem numa capital (65.25%)
- Predominância das Instituições sem acesso ao portal Capes (78.22%)
- Predominância das Instituições sem repetorio institucional (67.54%)
- Possuem busca integrada (60.80%)
- Possuem serviço de internet (83.84%)
- Participam em redes sociais (65.67%)
- Possuem predominâncai em ter outras bases (61.97%)
- Maioria das instituições são privada (com e sem fins lucrativos) (96,26%)
- Maioria dos anos da IES são Faculdade (99.72%)

- Pouca quantidade de técnicos abaixo da média (14.10)
- Quantidade de alunos média de alunos abaixo (282.62)
- Corpo docente muito pouco na média (17.92)
- Pouca quantidade de cursos (2.34)
- Despesas e receitas média baixa
- Quantidade de periodicos e livros eletronicos com relevância baixo

### Third cluster

In [189]:
third_cluster_agg = pd.read_csv('./data/Agglomerative/Terceiro agrupamento (Agglomerative).csv', sep=';')

In [190]:
get_total_categorical_values_cluster(third_cluster_agg)

====== NU_ANO_CENSO ======
2018: 528 ( 26.87%)
2019: 521 ( 26.51%)
2017: 464 ( 23.61%)
2016: 452 ( 23.00%)

====== REGIAO ======
Sudeste: 975 ( 49.62%)
Nordeste: 437 ( 22.24%)
Sul: 288 ( 14.66%)
Norte: 142 ( 7.23%)
Centro-Oeste: 123 ( 6.26%)

====== IN_CAPITAL ======
0: 1192 ( 60.66%)
1: 773 ( 39.34%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 1187 ( 60.41%)
1: 778 ( 39.59%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 1093 ( 55.62%)
0: 872 ( 44.38%)

====== IN_BUSCA_INTEGRADA ======
1: 1515 ( 77.10%)
0: 450 ( 22.90%)

====== IN_SERVICO_INTERNET ======
1: 1924 ( 97.91%)
0: 41 ( 2.09%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 1232 ( 62.70%)
0: 733 ( 37.30%)

====== IN_CATALOGO_ONLINE ======
1: 1703 ( 86.67%)
0: 262 ( 13.33%)

====== IN_OUTRAS_BASES ======
1.0: 1556 ( 79.19%)
0.0: 409 ( 20.81%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada sem fins lucrativos: 977 ( 49.72%)
Privada com fins lucrativos: 923 ( 46.97%)
Pública Municipal e Especial: 40 ( 2.04%)
Pública Federal: 15 (

In [191]:
third_cluster_agg[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      1,965.00                 1,965.00             1,965.00   1,965.00   
mean         151.54                22,190.13            16,930.37   5,742.87   
std          204.63               295,367.05           231,402.09  17,710.34   
min            1.00                     0.00                 0.00       2.00   
25%           51.00                    41.00             1,226.00   1,636.00   
50%           93.00                   320.00             6,623.00   3,059.00   
75%          182.00                 4,000.00             9,155.00   5,374.00   
max        3,482.00             6,437,318.00        10,000,000.00 370,652.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS    T1_VL_DESPESAS  
count     1,965.00  1,965.00         1,965.00          1,965.00  
mean        149.49     18.09    73,871,411.67     79,483,263.76  
std         111.51     16.45   303,142,944.33    676,157,147.30  
min           1.00      1.00        36,843.47      1,130,746.33  
25%          72.00      9.00    14,606,845.09     12,133,038.72  
50%         117.00     14.00    28,874,918.03     24,181,059.20  
75%         191.00     22.00    62,951,334.00     52,121,930.28  
max         807.00    183.00 6,258,050,290.00 22,161,393,800.00

Conclusões:

- Predominância das regiões Sudeste e Nordeste (49.62% e 22.24%)
- Predominãncia das IES não estarem numa capital (60.66%)
- Instituições sem acesso ao portal Capes (60.41%)
- Possuem predominância em possuir busca integrada (77.10%)
- Possuem predominância em serviço de internet (97.91%)
- Predominância em participar em redes sociais (62.70%)
- Possuem catalogo online (86.67%)
- Possuem predominância em ter outras bases (79.19%)
- Maioria das instituições são privada (com e sem fins lucrativos) (96,69%)
- Maioria dos anos da IES são Faculdade e centro universitario (67.23% e 30.59%)

- Quantidade média de técnicos moderada (151.54)
- Quantidade de alunos média de alunos normal (5,742.87	)
- Corpo docente com quantidade média moderada (149.49)
- Pouca quantidade de cursos (18.09)
- Despesas e receitas média moderadas para alto
- Quantidade de periodicos e livros eletronicos com relevância alto

### Forth cluster

In [192]:
forth_cluster_agg = pd.read_csv('./data/Agglomerative/Quarto agrupamento (Agglomerative).csv', sep=';')

In [193]:
get_total_categorical_values_cluster(forth_cluster_agg)

====== NU_ANO_CENSO ======
2019: 475 ( 32.03%)
2018: 417 ( 28.12%)
2017: 297 ( 20.03%)
2016: 294 ( 19.82%)

====== REGIAO ======
Sudeste: 496 ( 33.45%)
Sul: 357 ( 24.07%)
Nordeste: 304 ( 20.50%)
Centro-Oeste: 226 ( 15.24%)
Norte: 100 ( 6.74%)

====== IN_CAPITAL ======
0: 938 ( 63.25%)
1: 545 ( 36.75%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 958 ( 64.60%)
1: 525 ( 35.40%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
0: 764 ( 51.52%)
1: 719 ( 48.48%)

====== IN_BUSCA_INTEGRADA ======
1: 1083 ( 73.03%)
0: 400 ( 26.97%)

====== IN_SERVICO_INTERNET ======
1: 1431 ( 96.49%)
0: 52 ( 3.51%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 1043 ( 70.33%)
0: 440 ( 29.67%)

====== IN_CATALOGO_ONLINE ======
1: 1124 ( 75.79%)
0: 359 ( 24.21%)

====== IN_OUTRAS_BASES ======
1.0: 1024 ( 69.05%)
0.0: 459 ( 30.95%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada com fins lucrativos: 867 ( 58.46%)
Privada sem fins lucrativos: 582 ( 39.24%)
Pública Municipal e Especial: 31 ( 2.09%)
Pública Federal: 3 ( 0.

In [194]:
forth_cluster_agg[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      1,483.00                 1,483.00             1,483.00   1,483.00   
mean          24.47                17,567.76            12,180.19     629.78   
std           21.10               522,594.48           162,783.38     585.06   
min            1.00                     0.00                 0.00      11.00   
25%           11.00                    20.00             1,741.00     224.00   
50%           18.00                   100.00             6,372.00     475.00   
75%           30.00                   600.00             8,648.50     839.00   
max          311.00            20,083,543.00         6,229,914.00   4,488.00   

       QT_DOCENTES  QT_CURSO  T1_VL_RECEITAS  T1_VL_DESPESAS  
count     1,483.00  1,483.00        1,483.00        1,483.00  
mean         32.80      4.70    4,533,739.31    4,080,641.43  
std          20.75      3.47    6,067,438.10    5,209,064.96  
min           3.00      1.00        3,125.00       42,500.00  
25%          17.00      2.00    1,242,556.92    1,348,383.59  
50%          28.00      4.00    2,751,112.26    2,675,040.99  
75%          43.00      6.00    5,580,204.38    5,133,079.24  
max         155.00     29.00   79,434,370.49   79,923,866.93

Conclusões:

- Predominância dos anos 2018 e 2019 (60,15%)
- Predominância das regiões Sudeste e Sul (33.45%, 24.07%)
- Uma incidência das IES não estarem numa capital (63.25%)
- Predominância das Instituições sem acesso ao portal Capes ( 64.60%)
- Possuem predominância em busca integrada (73.03%)
- Possuem serviço de internet (96.49%)
- Predominância em participar em redes sociais (70.33%)
- Possuem predominância em ter catalogo online (75.79%)
- Possuem predominância em ter outras bases (69.05%)
- Maioria das iinstituições são privada (com e sem fins lucrativos) (97,7%)
- Maioria dos anos da IES são Faculdade (99.60%)

- Quantidade de técnicos na média baixo (24.47%)
- Quantidade de alunos média de alunos  moderado (629.78)
- Corpo docente abaixo da média  (32.80)
- Pouca quantidade média de cursos (4.70)
- Despesas e receitas média moderado
- Quantidade de periodicos e livros eletronicos com relevância alto

### Firth cluster

In [195]:
firth_cluster_agg = pd.read_csv('./data/Agglomerative/Quinto agrupamento (Agglomerative).csv', sep=';')

In [196]:
get_total_categorical_values_cluster(firth_cluster_agg)

====== NU_ANO_CENSO ======
2016: 321 ( 35.24%)
2017: 248 ( 27.22%)
2018: 214 ( 23.49%)
2019: 128 ( 14.05%)

====== REGIAO ======
Sudeste: 445 ( 48.85%)
Nordeste: 147 ( 16.14%)
Centro-Oeste: 134 ( 14.71%)
Sul: 124 ( 13.61%)
Norte: 61 ( 6.70%)

====== IN_CAPITAL ======
0: 634 ( 69.59%)
1: 277 ( 30.41%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 646 ( 70.91%)
1: 265 ( 29.09%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
0: 572 ( 62.79%)
1: 339 ( 37.21%)

====== IN_BUSCA_INTEGRADA ======
1: 635 ( 69.70%)
0: 276 ( 30.30%)

====== IN_SERVICO_INTERNET ======
1: 834 ( 91.55%)
0: 77 ( 8.45%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 618 ( 67.84%)
0: 293 ( 32.16%)

====== IN_CATALOGO_ONLINE ======
1: 665 ( 73.00%)
0: 246 ( 27.00%)

====== IN_OUTRAS_BASES ======
0.0: 556 ( 61.03%)
1.0: 355 ( 38.97%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada sem fins lucrativos: 548 ( 60.15%)
Privada com fins lucrativos: 294 ( 32.27%)
Pública Municipal e Especial: 48 ( 5.27%)
Pública Estadual: 16 ( 1.76%)

In [197]:
firth_cluster_agg[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count        911.00                   911.00               911.00     911.00   
mean          52.92                   166.28                55.79   1,429.16   
std          110.52                   963.53               433.94   1,203.80   
min            5.00                     0.00                 0.00      89.00   
25%           24.00                     0.00                 0.00     581.50   
50%           37.00                    14.00                 0.00   1,043.00   
75%           57.50                   100.00                 1.00   1,902.00   
max        2,207.00                17,156.00             8,000.00   8,367.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS  T1_VL_DESPESAS  
count       911.00    911.00           911.00          911.00  
mean         65.70      7.29    18,623,640.27   10,655,911.89  
std          41.84      4.95   136,380,970.82   15,830,657.17  
min           6.00      1.00        89,327.57      306,039.11  
25%          36.00      4.00     4,129,303.48    4,025,294.12  
50%          56.00      6.00     8,013,288.00    7,109,575.50  
75%          84.00     10.00    15,222,986.02   12,697,594.68  
max         300.00     30.00 4,000,452,318.90  294,064,951.75

Conclusões:

- Predominancia dos anos de 2016 e 2017 (35.24% e 27.22%)
- Predominância das regiões Sudeste e Nordeste e Centro-Oeste (48.85%, 16.14%, 14.71%)
- Predominãncia das IES não estarem numa capital (69.59%)
- Instituições sem acesso ao portal Capes (70.91%)
- Instituições sem acesso repositorio institucional (62.79%)
- Possuem predominância em possuir busca integrada (69.70%)
- Possuem predominância em serviço de internet (91.55%)
- Predominância em participar em redes sociais (67.84%)
- Possuem catalogo online (73.00%)
- Possuem predominância em ter outras bases (61.03%)
- Maioria das instituições são privada (com e sem fins lucrativos) (92,42%)
- Maioria dos anos da IES são Faculdade  (97.15%)

- Quantidade média de técnicos abaixo (52.92)
- Quantidade de alunos média de alunos normal (1,429.16)
- Corpo docente com quantidade média moderada (65.70)
- Pouca quantidade de cursos (7.29)
- Despesas e receitas média moderadas para alto
- Quantidade de periodicos e livros eletronicos com relevância baixo

# K-means

### First cluster

In [198]:
first_cluster_kmeans = pd.read_csv('./data/K-means/Primeiro agrupamento (K-means).csv', sep=';')

In [199]:
get_total_categorical_values_cluster(first_cluster_kmeans)

====== NU_ANO_CENSO ======
2016: 149 ( 27.75%)
2017: 137 ( 25.51%)
2018: 132 ( 24.58%)
2019: 119 ( 22.16%)

====== REGIAO ======
Sudeste: 201 ( 37.43%)
Nordeste: 120 ( 22.35%)
Sul: 101 ( 18.81%)
Norte: 71 ( 13.22%)
Centro-Oeste: 44 ( 8.19%)

====== IN_CAPITAL ======
0: 321 ( 59.78%)
1: 216 ( 40.22%)

====== IN_ACESSO_PORTAL_CAPES ======
1: 469 ( 87.34%)
0: 68 ( 12.66%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 271 ( 50.47%)
0: 266 ( 49.53%)

====== IN_BUSCA_INTEGRADA ======
1: 313 ( 58.29%)
0: 224 ( 41.71%)

====== IN_SERVICO_INTERNET ======
1: 510 ( 94.97%)
0: 27 ( 5.03%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 384 ( 71.51%)
0: 153 ( 28.49%)

====== IN_CATALOGO_ONLINE ======
1: 463 ( 86.22%)
0: 74 ( 13.78%)

====== IN_OUTRAS_BASES ======
0.0: 281 ( 52.33%)
1.0: 256 ( 47.67%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Pública Federal: 264 ( 49.16%)
Privada sem fins lucrativos: 123 ( 22.91%)
Pública Estadual: 113 ( 21.04%)
Privada com fins lucrativos: 24 ( 4.47%)
Pública Munic

In [200]:
first_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count        537.00                   537.00               537.00     537.00   
mean         765.03                    88.07             4,257.21  12,553.95   
std          771.47                 1,633.33            18,064.84  11,019.01   
min           20.00                     0.00                 0.00     375.00   
25%          290.00                     0.00                 0.00   5,038.00   
50%          557.00                     0.00                 1.00   9,090.00   
75%          935.00                     3.00             2,127.00  15,797.00   
max        5,670.00                37,835.00           223,350.00  68,600.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS    T1_VL_DESPESAS  
count       537.00    537.00           537.00            537.00  
mean        761.73     52.45   340,006,634.24    373,171,331.44  
std         657.55     40.42   550,425,214.82    878,480,744.42  
min          51.00      1.00       102,665.47     12,814,291.76  
25%         301.00     27.00    71,049,153.95     97,099,748.00  
50%         545.00     42.00   207,574,418.88    220,906,797.80  
75%         951.00     63.00   366,195,544.48    366,195,544.48  
max       3,701.00    321.00 7,599,192,871.00 15,890,130,144.40

Conclusões:

- Predominância das regiões Sudeste, Nordeste e Sul (37.43%, 22.35%, 18.81%)
- Uma incidência das IES não estarem numa capital (59,78%)
- Instituições com acesso ao portal Capes (87.34%)
- Possuem serviço de internet (94.97%)
- Participam em redes sociais (71.51%)
- Possuem catalogo online (86.22%)
- Quase a métade das IES são públicas federais, Privada sem fins lucrativos, Pública Estadual (49.16%, 22.91%, 21.04%)
- Mais da metade das IES são do tipo universidade e Instituto Federal de Educação, Ciência e Tecnologia (59.96%, 20.48%)

- Quantidade alta de técnicos na média (765.03)
- Quantidade de alunos média de alunos alto (12,553.95)
- Corpo docente forte na média (761.73)
- Quantidade considerável de cursos na média (média de 52.45)
- Despesas e receitas média moderadas
- Quantidade de periodicos e livros eletronicos com relevância baixa

### Second cluster

In [201]:
second_cluster_kmeans = pd.read_csv('./data/K-means/Segundo agrupamento (K-means).csv', sep=';')

In [202]:
get_total_categorical_values_cluster(second_cluster_kmeans)

====== NU_ANO_CENSO ======
2019: 297 ( 32.00%)
2018: 261 ( 28.12%)
2016: 195 ( 21.01%)
2017: 175 ( 18.86%)

====== REGIAO ======
Sudeste: 325 ( 35.02%)
Nordeste: 238 ( 25.65%)
Sul: 166 ( 17.89%)
Centro-Oeste: 153 ( 16.49%)
Norte: 46 ( 4.96%)

====== IN_CAPITAL ======
0: 551 ( 59.38%)
1: 377 ( 40.62%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 587 ( 63.25%)
1: 341 ( 36.75%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 477 ( 51.40%)
0: 451 ( 48.60%)

====== IN_BUSCA_INTEGRADA ======
1: 690 ( 74.35%)
0: 238 ( 25.65%)

====== IN_SERVICO_INTERNET ======
1: 863 ( 93.00%)
0: 65 ( 7.00%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 687 ( 74.03%)
0: 241 ( 25.97%)

====== IN_CATALOGO_ONLINE ======
1: 602 ( 64.87%)
0: 326 ( 35.13%)

====== IN_OUTRAS_BASES ======
1.0: 643 ( 69.29%)
0.0: 285 ( 30.71%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada com fins lucrativos: 624 ( 67.24%)
Privada sem fins lucrativos: 291 ( 31.36%)
Pública Estadual: 5 ( 0.54%)
Pública Municipal e Especial: 5 ( 0.54%)
P

In [203]:
second_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count        928.00                   928.00               928.00     928.00   
mean          12.87                 1,680.80            11,675.25     153.83   
std           13.90                 8,124.14           210,971.48     154.13   
min            1.00                     0.00                 0.00       1.00   
25%            6.00                    20.00               200.00      53.75   
50%            9.00                    70.00             2,100.00     113.00   
75%           15.00                   400.00             7,726.75     215.50   
max          161.00               179,324.00         6,424,945.00   2,086.00   

       QT_DOCENTES  QT_CURSO  T1_VL_RECEITAS  T1_VL_DESPESAS  
count       928.00    928.00          928.00          928.00  
mean         14.20      2.30    1,475,556.24    1,626,618.40  
std          10.38      1.94   15,700,553.63   15,868,979.79  
min           1.00      1.00            0.01            2.00  
25%           9.00      1.00      211,203.48      262,411.89  
50%          12.00      2.00      615,428.62      629,806.48  
75%          18.00      3.00    1,157,021.17    1,238,916.58  
max         154.00     24.00  476,631,170.46  481,208,448.88

Conclusões:

- Predominância dos anos 2018 e 2019 (60,12%)
- Predominância das regiões Sudeste e Nordeste (60,67%)
- Uma incidência das IES não estarem numa capital (59.38%)
- Predominância das Instituições sem acesso ao portal Capes (63.25%)
- Possuem busca integrada (74.35%)
- Possuem serviço de internet (93.00%)
- Participam em redes sociais (74.03%)
- Possuem predominância em catalogo online (64.87%)
- Possuem predominâncai em ter outras bases (69.29%)
- Maioria das instituições são privada (com e sem fins lucrativos) (98,6)
- Maioria dos anos da IES são Faculdade (99.89%)

- Pouca quantidade de técnicos na média (12.87)
- Quantidade de alunos média de alunos abaixo (153.83)
- Corpo docente muito pouco na média (14.20)
- Pouca quantidade de cursos (2.30)
- Despesas e receitas média moderadas
- Quantidade de periodicos e livros eletronicos com relevância baixa

### Third cluster

In [204]:
third_cluster_kmeans = pd.read_csv('./data/K-means/Terceiro agrupamento (K-means).csv', sep=';')

In [205]:
get_total_categorical_values_cluster(third_cluster_kmeans)

====== NU_ANO_CENSO ======
2016: 345 ( 27.91%)
2017: 337 ( 27.27%)
2018: 286 ( 23.14%)
2019: 268 ( 21.68%)

====== REGIAO ======
Sudeste: 583 ( 47.17%)
Nordeste: 237 ( 19.17%)
Sul: 216 ( 17.48%)
Centro-Oeste: 112 ( 9.06%)
Norte: 88 ( 7.12%)

====== IN_CAPITAL ======
0: 775 ( 62.70%)
1: 461 ( 37.30%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 1027 ( 83.09%)
1: 209 ( 16.91%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
0: 923 ( 74.68%)
1: 313 ( 25.32%)

====== IN_BUSCA_INTEGRADA ======
1: 651 ( 52.67%)
0: 585 ( 47.33%)

====== IN_SERVICO_INTERNET ======
1: 978 ( 79.13%)
0: 258 ( 20.87%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 774 ( 62.62%)
0: 462 ( 37.38%)

====== IN_CATALOGO_ONLINE ======
0: 668 ( 54.05%)
1: 568 ( 45.95%)

====== IN_OUTRAS_BASES ======
0.0: 934 ( 75.57%)
1.0: 302 ( 24.43%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada sem fins lucrativos: 643 ( 52.02%)
Privada com fins lucrativos: 565 ( 45.71%)
Pública Municipal e Especial: 21 ( 1.70%)
Pública Estadual: 6 ( 0.49%

In [206]:
third_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      1,236.00                 1,236.00             1,236.00   1,236.00   
mean          10.78                    38.75                17.38     185.81   
std           13.17                   390.28               177.16     160.48   
min            1.00                     0.00                 0.00       1.00   
25%            5.00                     0.00                 0.00      70.00   
50%            8.00                     0.00                 0.00     143.00   
75%           13.00                    10.00                 0.00     260.25   
max          277.00                 9,331.00             4,000.00   1,587.00   

       QT_DOCENTES  QT_CURSO  T1_VL_RECEITAS  T1_VL_DESPESAS  
count     1,236.00  1,236.00        1,236.00        1,236.00  
mean         16.72      2.03    1,550,846.21    1,585,898.44  
std          10.01      1.39    3,915,629.37    6,214,497.12  
min           1.00      1.00            0.01            2.00  
25%          10.00      1.00      389,521.86      387,356.53  
50%          15.00      2.00      855,917.78      855,450.93  
75%          22.00      3.00    1,559,343.44    1,567,425.95  
max         142.00     20.00   78,935,409.20  191,892,519.06

Conclusões:

- Predominância das regiões Sudeste e Nordeste e Sul (47.17%, 19.17%, 17.48%)
- Predominãncia das IES não estarem numa capital (62.70%)
- Instituições sem acesso ao portal Capes (83.09%)
- Possuem predominância em não possuir repositorio institucional (74.68%)
- Possuem predominância em serviço de internet (79.13%)
- Predominância em participar em redes sociais (62.62%)
- Possuem predominância em ter outras bases (75.57%)
- Maioria das instituições são privada (com e sem fins lucrativos) (97,73)
- Maioria dos anos da IES são Faculdade (99.92%)

- Quantidade baixa de média de técnicos (10.78)
- Quantidade de alunos média de alunos baixo (185.81	)
- Corpo docente muito baixo na média (16.72)
- Pouca quantidade de cursos (2.03)
- Despesas e receitas média moderadas
- Quantidade de periodicos e livros eletronicos com relevância baixa

### Forth cluster

In [207]:
forth_cluster_kmeans = pd.read_csv('./data/K-means/Quarto agrupamento (K-means).csv', sep=';')

In [208]:
get_total_categorical_values_cluster(forth_cluster_kmeans)

====== NU_ANO_CENSO ======
2019: 468 ( 30.63%)
2018: 450 ( 29.45%)
2017: 318 ( 20.81%)
2016: 292 ( 19.11%)

====== REGIAO ======
Sudeste: 596 ( 39.01%)
Sul: 346 ( 22.64%)
Nordeste: 325 ( 21.27%)
Centro-Oeste: 159 ( 10.41%)
Norte: 102 ( 6.68%)

====== IN_CAPITAL ======
0: 1052 ( 68.85%)
1: 476 ( 31.15%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 989 ( 64.73%)
1: 539 ( 35.27%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 808 ( 52.88%)
0: 720 ( 47.12%)

====== IN_BUSCA_INTEGRADA ======
1: 1181 ( 77.29%)
0: 347 ( 22.71%)

====== IN_SERVICO_INTERNET ======
1: 1489 ( 97.45%)
0: 39 ( 2.55%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 1088 ( 71.20%)
0: 440 ( 28.80%)

====== IN_CATALOGO_ONLINE ======
1: 1214 ( 79.45%)
0: 314 ( 20.55%)

====== IN_OUTRAS_BASES ======
1.0: 1122 ( 73.43%)
0.0: 406 ( 26.57%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada com fins lucrativos: 866 ( 56.68%)
Privada sem fins lucrativos: 632 ( 41.36%)
Pública Municipal e Especial: 28 ( 1.83%)
Pública Estadual: 2 ( 

In [209]:
forth_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      1,528.00                 1,528.00             1,528.00   1,528.00   
mean          33.68                17,212.81            19,758.56     932.89   
std           26.16               514,813.19           301,869.62     858.73   
min            2.00                     0.00                 5.00      22.00   
25%           17.00                    29.00             2,000.00     487.00   
50%           27.00                   129.00             7,000.00     771.50   
75%           41.00                   559.25             8,734.00   1,205.25   
max          346.00            20,083,543.00        10,000,000.00  24,310.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS   T1_VL_DESPESAS  
count     1,528.00  1,528.00         1,528.00         1,528.00  
mean         45.12      6.21    16,180,821.28    11,879,650.46  
std          22.36      4.01   186,450,066.90    87,466,525.39  
min           1.00      1.00         3,125.00       248,908.99  
25%          29.00      3.00     2,793,845.86     2,633,540.37  
50%          41.00      5.00     5,174,117.05     4,541,098.14  
75%          58.00      8.00     9,047,620.84     7,993,876.03  
max         153.00     40.00 5,745,640,000.00 2,918,280,000.00

Conclusões:

- Predominância dos anos 2018 e 2019 (60,08%)
- Predominância das regiões Sudeste, Sul e Nordeste (82,92%)
- Uma incidência das IES não estarem numa capital (68.85%)
- Predominância das Instituições sem acesso ao portal Capes ( 64.73%)
- Possuem predominância em busca integrada (77.29%)
- Possuem serviço de internet (97.45%)
- Predominância em participar em redes sociais (71.20%)
- Possuem predominância em ter catalogo online (79.45%)
- Possuem predominância em ter outras bases (73.43%)
- Maioria das iinstituições são privada (com e sem fins lucrativos) (98,04)
- Maioria dos anos da IES são Faculdade (99.15%)

- Quantidade de técnicos na média baixo (33.68	)
- Quantidade de alunos média de alunos  moderado (932.89)
- Corpo docente muito na média (45.12)
- Pouca quantidade média de cursos (6.21)
- Despesas e receitas média acima do moderado
- Quantidade de periodicos e livros eletronicos com relevância alto

### Firth cluster

In [210]:
firth_cluster_kmeans = pd.read_csv('./data/K-means/Quinto agrupamento (K-means).csv', sep=';')

In [211]:
get_total_categorical_values_cluster(firth_cluster_kmeans)

====== NU_ANO_CENSO ======
2019: 347 ( 26.63%)
2018: 334 ( 25.63%)
2017: 322 ( 24.71%)
2016: 300 ( 23.02%)

====== REGIAO ======
Sudeste: 571 ( 43.82%)
Nordeste: 334 ( 25.63%)
Sul: 189 ( 14.50%)
Norte: 109 ( 8.37%)
Centro-Oeste: 100 ( 7.67%)

====== IN_CAPITAL ======
0: 771 ( 59.17%)
1: 532 ( 40.83%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 828 ( 63.55%)
1: 475 ( 36.45%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 718 ( 55.10%)
0: 585 ( 44.90%)

====== IN_BUSCA_INTEGRADA ======
1: 1015 ( 77.90%)
0: 288 ( 22.10%)

====== IN_SERVICO_INTERNET ======
1: 1270 ( 97.47%)
0: 33 ( 2.53%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 787 ( 60.40%)
0: 516 ( 39.60%)

====== IN_CATALOGO_ONLINE ======
1: 1115 ( 85.57%)
0: 188 ( 14.43%)

====== IN_OUTRAS_BASES ======
1.0: 1020 ( 78.28%)
0.0: 283 ( 21.72%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada com fins lucrativos: 715 ( 54.87%)
Privada sem fins lucrativos: 547 ( 41.98%)
Pública Municipal e Especial: 27 ( 2.07%)
Pública Federal: 9 ( 0.69

In [212]:
firth_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      1,303.00                 1,303.00             1,303.00   1,303.00   
mean         143.65                31,306.84            11,317.78   4,691.86   
std          136.18               362,333.87            55,653.40   4,874.23   
min           12.00                     0.00                 0.00      72.00   
25%           70.00                   105.00               564.00   2,496.00   
50%          108.00                   422.00             6,157.00   3,621.00   
75%          180.50                 5,625.00             8,729.50   5,386.50   
max        2,654.00             6,437,318.00         1,200,000.00  89,706.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS    T1_VL_DESPESAS  
count     1,303.00  1,303.00         1,303.00          1,303.00  
mean        151.18     18.49    52,263,088.00     61,435,145.70  
std          75.60     13.72   140,361,418.17    529,816,318.16  
min          10.00      1.00       312,066.18      2,600,437.90  
25%          97.00     11.00    20,850,115.51     16,877,562.19  
50%         132.00     16.00    32,806,982.59     27,917,450.09  
75%         189.00     22.00    59,547,232.10     48,137,697.00  
max         550.00    183.00 4,158,885,757.32 17,002,300,000.00

Conclusões:

- Predominância das regiões Sudeste e Nordeste (69,45%)
- Uma incidência das IES não estarem numa capital (59.17%)
- Predominância das Instituições sem acesso ao portal Capes ( 63.55%)
- Possuem predominância em busca integrada (77.90%)
- Possuem serviço de internet (97.47%)
- Predominância em participar em redes sociais (60.40%)
- Possuem catalogo online (85.57%)
- Possuem predominância em ter outras bases (78.28%)
- Maioria das instituições são privada (com e sem fins lucrativos) (96,85)
- Maioria dos anos da IES são Faculdade e centro universitário (66.16%, 32.23%)

- Quantidade de técnicos médio moderado (143.65)
- Quantidade de alunos média de alunos acima do moderado (4,691.86)
- Corpo docente muito acima da  média (151.18)
- Moderada quantidade média de cursos (18.49)
- Despesas e receitas média altas
- Quantidade de periodicos e livros eletronicos com relevância alto

### Sixth cluster

In [213]:
sixth_cluster_kmeans = pd.read_csv('./data/K-means/Sexto agrupamento (K-means).csv', sep=';')

In [214]:
get_total_categorical_values_cluster(sixth_cluster_kmeans)

====== NU_ANO_CENSO ======
2016: 471 ( 35.63%)
2017: 372 ( 28.14%)
2018: 261 ( 19.74%)
2019: 218 ( 16.49%)

====== REGIAO ======
Sudeste: 577 ( 43.65%)
Nordeste: 275 ( 20.80%)
Sul: 217 ( 16.41%)
Centro-Oeste: 159 ( 12.03%)
Norte: 94 ( 7.11%)

====== IN_CAPITAL ======
0: 915 ( 69.21%)
1: 407 ( 30.79%)

====== IN_ACESSO_PORTAL_CAPES ======
0: 967 ( 73.15%)
1: 355 ( 26.85%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
0: 899 ( 68.00%)
1: 423 ( 32.00%)

====== IN_BUSCA_INTEGRADA ======
1: 865 ( 65.43%)
0: 457 ( 34.57%)

====== IN_SERVICO_INTERNET ======
1: 1198 ( 90.62%)
0: 124 ( 9.38%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 863 ( 65.28%)
0: 459 ( 34.72%)

====== IN_CATALOGO_ONLINE ======
1: 957 ( 72.39%)
0: 365 ( 27.61%)

====== IN_OUTRAS_BASES ======
0.0: 823 ( 62.25%)
1.0: 499 ( 37.75%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada sem fins lucrativos: 667 ( 50.45%)
Privada com fins lucrativos: 523 ( 39.56%)
Pública Municipal e Especial: 100 ( 7.56%)
Pública Estadual: 24 ( 1.8

In [215]:
sixth_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count      1,322.00                 1,322.00             1,322.00   1,322.00   
mean          46.05                   181.31                20.60   1,202.21   
std           95.21                 1,624.34               157.65   1,028.28   
min            1.00                     0.00                 0.00      62.00   
25%           19.00                     0.00                 0.00     503.25   
50%           31.00                     6.00                 0.00     879.00   
75%           50.00                    63.75                 0.00   1,503.25   
max        2,207.00                38,000.00             3,090.00   8,316.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS   T1_VL_DESPESAS  
count     1,322.00  1,322.00         1,322.00         1,322.00  
mean         60.92      6.76    20,169,429.75    12,427,568.59  
std          41.19      5.27   174,260,340.20    75,731,466.95  
min           3.00      1.00             1.00        98,800.00  
25%          34.00      4.00     3,080,043.69     3,007,529.00  
50%          50.00      6.00     6,047,025.04     5,573,239.50  
75%          73.00      9.00    12,369,185.83    11,492,529.14  
max         300.00    113.00 4,000,452,318.90 2,255,916,597.00

Conclusões:

- Predominância dos anos 2016 e 2017 (35.63%, 28.14%)
- Predominância das regiões Sudeste e Nordeste (43.65%, 20.80%)
- Predominância das IES não estarem numa capital (69.21%)
- Predominância das Instituições sem acesso ao portal Capes (73.15%)
- Predominância das Instituições sem repositorio institucional (68.00%)
- Possuem predominância em busca integrada (65.43%)
- Possuem serviço de internet (90.62%)
- Predominância em participar em redes sociais (65.28%)
- Possuem catalogo online (72.39%)
- Possuem predominância em ter outras bases (62.25%)
- Maioria das instituições são privada (com e sem fins lucrativos) (90,01%)
- Maioria dos anos da IES são Faculdade (95.39%)

- Quantidade de técnicos médio baixo (46.05	)
- Quantidade de alunos média de alunos  moderado (1,202.21)
- Corpo docente moderado na  média (60.92)
- Moderada quantidade baixa de cursos (16.76)
- Despesas e receitas média altas
- Baixa quantidade de Quantidade de periodicos e livros eletronicos
- Quantidade de periodicos e livros eletronicos com relevância baixo

### Seventh cluster

In [216]:
seventh_cluster_kmeans = pd.read_csv('./data/K-means/Sétimo agrupamento (K-means).csv', sep=';')

In [217]:
get_total_categorical_values_cluster(seventh_cluster_kmeans)

====== NU_ANO_CENSO ======
2019: 161 ( 27.38%)
2018: 149 ( 25.34%)
2017: 145 ( 24.66%)
2016: 133 ( 22.62%)

====== REGIAO ======
Sudeste: 264 ( 44.90%)
Sul: 143 ( 24.32%)
Nordeste: 108 ( 18.37%)
Centro-Oeste: 48 ( 8.16%)
Norte: 25 ( 4.25%)

====== IN_CAPITAL ======
1: 317 ( 53.91%)
0: 271 ( 46.09%)

====== IN_ACESSO_PORTAL_CAPES ======
1: 516 ( 87.76%)
0: 72 ( 12.24%)

====== IN_REPOSITORIO_INSTITUCIONAL ======
1: 414 ( 70.41%)
0: 174 ( 29.59%)

====== IN_BUSCA_INTEGRADA ======
1: 395 ( 67.18%)
0: 193 ( 32.82%)

====== IN_SERVICO_INTERNET ======
1: 581 ( 98.81%)
0: 7 ( 1.19%)

====== IN_PARTICIPA_REDE_SOCIAL ======
1: 423 ( 71.94%)
0: 165 ( 28.06%)

====== IN_CATALOGO_ONLINE ======
1: 545 ( 92.69%)
0: 43 ( 7.31%)

====== IN_OUTRAS_BASES ======
1.0: 536 ( 91.16%)
0.0: 52 ( 8.84%)

====== NOME_CATEGORIA_ADMINISTRATIVA ======
Privada sem fins lucrativos: 272 ( 46.26%)
Pública Federal: 151 ( 25.68%)
Privada com fins lucrativos: 117 ( 19.90%)
Pública Estadual: 37 ( 6.29%)
Pública Municipal 

In [218]:
seventh_cluster_kmeans[numeric_columns].describe()

QT_TEC_TOTAL  QT_PERIODICO_ELETRONICO  QT_LIVRO_ELETRONICO  QT_ALUNOS  \
count        588.00                   588.00               588.00     588.00   
mean       1,170.62                20,580.04            51,040.33  27,666.31   
std        1,577.99                45,203.40           117,262.32  37,426.76   
min           42.00                     0.00                 0.00     933.00   
25%          417.75                   354.50             5,305.00  10,074.00   
50%          727.50                 5,416.50            10,810.50  18,228.00   
75%        1,241.50                27,972.00            21,087.00  32,646.50   
max       15,562.00               572,574.00         1,000,000.00 370,652.00   

       QT_DOCENTES  QT_CURSO   T1_VL_RECEITAS    T1_VL_DESPESAS  
count       588.00    588.00           588.00            588.00  
mean        970.69     73.91   507,239,181.16    531,850,625.47  
std         897.50     48.37   726,312,568.20  1,146,523,723.87  
min          90.00     10.00       796,666.35     42,123,282.22  
25%         409.75     40.00   148,696,340.42    142,580,692.62  
50%         640.50     59.00   283,766,807.44    260,395,339.91  
75%       1,230.75     99.00   566,429,608.87    506,231,701.56  
max       6,286.00    296.00 6,258,050,290.00 22,161,393,800.00

Conclusões:

- Predominância das regiões Sudeste e Sul (44.90%, 24.32%)
- Predominância das Instituições com acesso ao portal Capes (87.76%)
- Predominância das Instituições com repositorio institucional (70.41%)
- Possuem predominância em busca integrada (67.18%)
- Possuem serviço de internet (98.81%)
- Predominância em participar em redes sociais (71.94%)
- Possuem catalogo online (92.69%)
- Possuem predominância em ter outras bases (91.16%)
- Maioria das instituições são privada sem fins, federal e com fins lurativos (46.26%, 25.68%, 19.90%)
- Maioria dos anos da IES são Universidade e centro universitario (69.73%, 22.28%)

- Quantidade de técnicos médio alto (1,170.62	)
- Quantidade de alunos média de alunos alto (27,666.31)
- Corpo docente alto na  média (970.69)
- altga quantidade baixa de cursos (73.91)
- Despesas e receitas média altas
- Alta quantidade de Quantidade de periodicos e livros eletronicos
- Quantidade de periodicos e livros eletronicos com relevância alto

# K-means

In [219]:
analise_kmeans = pd.DataFrame([], columns=['prompt', 'cluster', 'temperature', 'Total_suposições', 'Mapeadas_pelo_modelo', 'Ignoradas_pelo_modelo', 'alucinação_do_modelo', 'insights'])

In [220]:
def show_result_kmeans_prompt(df, temperature, cluster):
    return df.loc[(df['temperature'] == temperature) & (df['cluster'] == cluster), ['result']]['result'].tolist()[0]

In [221]:
kmeans_fase2.loc[(kmeans_fase2['temperature'] == 1) & (kmeans_fase2['cluster'] == 'primeiro'), ['result']]['result'].tolist()[0]

'Para iniciar nossa análise do cluster fornecido no CSV, é fundamental entender o conceito de cluster e como ele pode estar relacionado com as características dos dados. Um cluster é um grupo de pontos no espaço de dados que se agrupam de acordo com suas características e tendências em comum.\n\nAqui está a cadeia de pensamento para iniciar a análise:\n\n1. **Entendendo o conceito de cluster**: Antes de analisar os dados, é crucial entender o que um cluster significa e como ele pode ser aplicado à nossa análise.\n2. **Verificando a estrutura do dataset**: Ao examinar o CSV fornecido, precisamos verificar se há alguma estrutura ou padrão em como as características estão relacionadas entre si.\n3. **Identificando características relevantes**: Nossas primeiras etapas envolverão analisar cada característica isolada para determinar sua importância no agrupamento dos dados.\n4. **Conjunta análise de características**: A análise conjunta de características significa que vamos avaliar como as 

In [222]:
analise_kmeans

Empty DataFrame
Columns: [prompt, cluster, temperature, Total_suposições, Mapeadas_pelo_modelo, Ignoradas_pelo_modelo, alucinação_do_modelo, insights]
Index: []

## Fase 2

In [223]:
kmeans_fase2 = pd.read_csv('Models/webUI/results/K-means_Fase 2_results.csv')

### First Cluster

In [224]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'primeiro'))

Aqui está o meu processo de análise:

**Passo 1: Entendendo o Dados**

Ao receber o CSV contendo as instâncias do cluster, o primeiro passo é entender a natureza dos dados. Qual é o objetivo da análise? Quais características estamos considerando? Neste caso, não há informações sobre o objetivo ou as características específicas, então vou me concentrar em analisar os dados estatísticos.

**Passo 2: Visuais Iniciais**

Para obter uma visão geral dos agrupamentos, vou criar alguns gráficos simples. Vou plotar um gráfico de dispersão para visualizar as características principais do dataset. Isso ajudará a identificar quais características estão mais relacionadas entre si e se há algum padrão ou tendência.

**Gráfico de Dispersão**

A análise dos dados estatísticos sugere que o cluster tem uma distribuição simétrica para algumas características, enquanto outras têm uma distribuição não simétrica. O gráfico de dispersão sugere que as duas primeiras características são principais no agrupamen

Conclusões:

- Predominância das regiões Sudeste, Nordeste e Sul (37.43%, 22.35%, 18.81%) (NOT OK)
- Uma incidência das IES não estarem numa capital (59,78%) (NOT OK)
- Instituições com acesso ao portal Capes (87.34%) (NOT OK)
- Possuem serviço de internet (94.97%) (NOT OK)
- Participam em redes sociais (71.51%) (NOT OK)
- Possuem catalogo online (86.22%) (NOT OK)
- Busca integrada (58.29%) (NOT OK)
- Quase a métade das IES são públicas federais, Privada sem fins lucrativos, Pública Estadual (49.16%, 22.91%, 21.04%) (NOT OK)
- Mais da metade das IES são do tipo universidade e Instituto Federal de Educação, Ciência e Tecnologia (59.96%, 20.48%) (NOT OK)

- Quantidade alta de técnicos na média (765.03) (NOT OK)
- Quantidade de alunos média de alunos alto (12,553.95) (NOT OK)
- Corpo docente forte na média (761.73) (NOT OK)
- Quantidade considerável de cursos na média (média de 52.45) (NOT OK)
- Despesas e receitas média moderadas (NOT OK)
- Quantidade de periodicos e livros eletronicos com relevância baixa (NOT OK)

Insights falsos:

- Análise de gráficos que não são gerados
- Falta de explicação sobre as variaveis 
- Detalhamento do passo a passo ao invés da análise em si


Enfatiza a presença de analises mais detalhadas

In [225]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2', 'primeiro', 0, 15, 7, 8, 1, 0]

In [226]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'primeiro'))

**Análise do Cluster**

A análise do cluster será realizada com base em estatísticas e padrões observados nos dados fornecidos. O objetivo é identificar os fatores que levaram à criação desse agrupamento específico.

**Características Principais**

Ao analisar as instâncias do cluster, podemos observar as seguintes características principais:

* **Distribuição de Vários Attributos**: A distribuição dos atributos (como idade, salário e nível de educação) não é uniforme em todas as instâncias do cluster. Isso sugere que há uma correlação entre essas variáveis.
* **Concentração de Valor**: Algumas características apresentam valores concentrados em certos intervalos (por exemplo, um grupo de pessoas com salário extremamente alto ou baixo). Isso pode indicar uma tendência ou uma característica específica associada a esse cluster.

**Análise Estatística**

* **Medidas de Desvios**: A mediana da idade é 30 anos e o valor médio do salário é de R$ 10.000,00. Isso sugere que as pessoas nesse clu

Conclusões:

- Predominância das regiões Sudeste, Nordeste e Sul (37.43%, 22.35%, 18.81%) (NOT OK)
- Uma incidência das IES não estarem numa capital (59,78%) NOT OK
- Instituições com acesso ao portal Capes (87.34%) NOT OK
- Possuem serviço de internet (94.97%) NOT OK
- Participam em redes sociais (71.51%) NOT OK
- Possuem catalogo online (86.22%) NOT OK
- Busca integrada (58.29%) NOT OK
- Quase a métade das IES são públicas federais, Privada sem fins lucrativos, Pública Estadual (49.16%, 22.91%, 21.04%) NOT OK
- Mais da metade das IES são do tipo universidade e Instituto Federal de Educação, Ciência e Tecnologia (59.96%, 20.48%) NOT OK

- Quantidade alta de técnicos na média (765.03) NOTOK
- Quantidade de alunos média de alunos alto (12,553.95) NOTOK
- Corpo docente forte na média (761.73) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOT NOTOK

Insights falsos:

- A análise apresentou dados inconsistentes com o foco da analise(alucinou)


In [227]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'primeiro', 1, 15, 9, 6, 2, 0]

In [228]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'primeiro'))

Para iniciar nossa análise do cluster fornecido no CSV, é fundamental entender o conceito de cluster e como ele pode estar relacionado com as características dos dados. Um cluster é um grupo de pontos no espaço de dados que se agrupam de acordo com suas características e tendências em comum.

Aqui está a cadeia de pensamento para iniciar a análise:

1. **Entendendo o conceito de cluster**: Antes de analisar os dados, é crucial entender o que um cluster significa e como ele pode ser aplicado à nossa análise.
2. **Verificando a estrutura do dataset**: Ao examinar o CSV fornecido, precisamos verificar se há alguma estrutura ou padrão em como as características estão relacionadas entre si.
3. **Identificando características relevantes**: Nossas primeiras etapas envolverão analisar cada característica isolada para determinar sua importância no agrupamento dos dados.
4. **Conjunta análise de características**: A análise conjunta de características significa que vamos avaliar como as diferent

Conclusões:

- Predominância das regiões Sudeste, Nordeste e Sul (37.43%, 22.35%, 18.81%) OK
- Uma incidência das IES não estarem numa capital (59,78%) OK
- Instituições com acesso ao portal Capes (87.34%) OK
- Possuem serviço de internet (94.97%) OK
- Participam em redes sociais (71.51%) OK 
- Possuem catalogo online (86.22%) NOT OK
- Busca integrada (58.29%) NOT OK
- Quase a métade das IES são públicas federais, Privada sem fins lucrativos, Pública Estadual (49.16%, 22.91%, 21.04%) NOT OK
- Mais da metade das IES são do tipo universidade e Instituto Federal de Educação, Ciência e Tecnologia (59.96%, 20.48%) NOT OK

- Quantidade alta de técnicos na média (765.03) NOT OK
- Quantidade de alunos média de alunos alto (12,553.95) NOT OK
- Corpo docente forte na média (761.73) NOT OK
- Despesas e receitas média moderadas NOT OK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOT OK

Insights falsos:
- Há um alto número de instituições com repositório institucional (`1`), sugerindo prática comum de preservação e compartilhamento de dados e materiais de pesquisa, porém há algumas exceções que merecem investigação adicional.
- Variável não tao relevantes : **`IN_CATALOGO_ONLINE`**




In [229]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'primeiro', 2, 15, 5, 10, 2, 0]

### Second Cluster

In [230]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'segundo'))

Uma abordagem para entender o agrupamento dos dados em um cluster é analisar os padrões estatísticos e as relações entre as variáveis presentes nos dados.

**Primeiro Passo: Entendendo o Cluster**

Antes de começar a análise, é importante entender que um cluster é uma coleção de pontos próximos uns dos outros no espaço de dados. O objetivo aqui é identificar os fatores que levaram à formação desse agrupamento.

**Análise Estatística**

Para fazer isso, precisamos analisar as variáveis presentes nos dados e entender como elas interagem entre si. Vamos considerar algumas características importantes:

*   **Central Tendência**: a tendência média dos valores de cada característica.
*   **Variância**: a dispersão dos valores ao redor da tendência média.
*   **Correlação**: a relação linear ou não linear entre as variáveis.

Aqui estão alguns possíveis fatores que podem ter levado à criação do cluster:

1.  **Características Isoladas**: uma característica pode ser única e separada dos outros

Conclusões:

- Predominância dos anos 2018 e 2019 (60,12%) NOTOK
- Predominância das regiões Sudeste e Nordeste (60,67%) NOTOK
- Uma incidência das IES não estarem numa capital (59.38%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes (63.25%) NOTOK
- Possuem busca integrada (74.35%) OK
- Possuem serviço de internet (93.00%) NOTOK
- Participam em redes sociais (74.03%) NOTOK
- Possuem predominância em catalogo online (64.87%) NOTOK
- Possuem predominâncai em ter outras bases (69.29%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (98,6) OK
- Maioria dos anos da IES são Faculdade (99.89%) NOTOK

- Pouca quantidade de técnicos na média (12.87) NOTOK
- Quantidade de alunos média de alunos abaixo (153.83) NOTOK
- Corpo docente muito pouco na média (14.20) NOTOK
- Pouca quantidade de cursos (2.30) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOTOK

Insights falsos:
-  Mudança de contexto da análise, o modelo criou váriaveis que não existiam




In [231]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'segundo', 0, 17, 2, 15, 4, 0]

In [232]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'segundo'))

**Análise de Clustering**

A análise de clustering é um processo que visa identificar e entender agrupamentos significativos em dados estatísticos. Nesse contexto, recebi um conjunto de dados em formato CSV que representa instâncias de um determinado cluster. O objetivo é analisar essas instâncias e explicar a criação do cluster.

**Características do Dados**

Ao analisar o conjunto de dados, percebei que ele consiste em várias características estatísticas, incluindo:

*   Variável de Medida de Dispersão (VMD): uma medida que representa a espessura ou distância entre os pontos.
*   Vértice de Dispersion (VD): um valor que representa a coordenada x do vértice de dispersão.
*   Coordenada y da Mediana: uma medida que representa a posição média das instâncias em relação à variável de medida de dispersão.

**Análise de Clustering**

Ao analisar as instâncias, percebi que elas podem ser agrupadas em diferentes formas. A forma mais significativa é a criação de um cluster isolado na extremida

Conclusões:

- Predominância dos anos 2018 e 2019 (60,12%) NOTOK
- Predominância das regiões Sudeste e Nordeste (60,67%) NOTOK 
- Uma incidência das IES não estarem numa capital (59.38%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes (63.25%) NOTOK
- Possuem busca integrada (74.35%) NOTOK
- Possuem serviço de internet (93.00%) NOTOK
- Participam em redes sociais (74.03%) NOTOK
- Possuem predominância em catalogo online (64.87%) NOTOK
- Possuem predominâncai em ter outras bases (69.29%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (98,6) NOTOK
- Maioria dos anos da IES são Faculdade (99.89%) NOTOK

- Pouca quantidade de técnicos na média (12.87) NOTOK
- Quantidade de alunos média de alunos abaixo (153.83) NOTOK
- Corpo docente muito pouco na média (14.20) NOTOK
- Pouca quantidade de cursos (2.30) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOTOK

Insights falsos:
- a maioria possui repositórios institucionais (`IN_REPOSITORIO_INSTITUCIONAL`)
- Observa-se que as variáveis relacionadas a outros recursos digitais (busca integrada, redes sociais, catálogos online e outras bases de dados) mostram uma variabilidade maior, sem um padrão claro.
- As instituições estão distribuídas por diversas regiões do país, sem nenhuma região predominante.

In [233]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'segundo', 1, 17, 4, 13, 2, 0]

In [234]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'segundo'))

**Análise do Cluster**

Recebido o arquivo CSV com as instâncias do cluster, farei uma análise estatística detalhada para entender os fatores que levaram à criação desse agrupamento.

**Etapas da Análise**

1. **Visualização Inicial**: Comecei visualizando a distribuição das instâncias no espaço de características. Isso me permitiu identificar rapidamente se o cluster era isolado ou se existia algum padrão de comportamento interessante.
2. **Análise das Medidas Centrícias**: Aproximadamente 70% das instâncias se concentravam em um único quadrante do espaço de características, sugerindo que houve uma relação forte entre as características.
3. **Comparação entre Categorias de Variações**: Ao comparar as variações entre os diferentes quadrantes do espaço de características, observei que as primeiras três características (X1, X2 e X3) apresentavam uma grande variabilidade nos valores da classe, enquanto as últimas duas características (X4 e X5) tinham valores mais estacionários.
4. **Análi

Conclusões:

- Predominância dos anos 2018 e 2019 (60,12%) NOTOK
- Predominância das regiões Sudeste e Nordeste (60,67%) NOTOK
- Uma incidência das IES não estarem numa capital (59.38%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes (63.25%) NOTOK
- Possuem busca integrada (74.35%) OK
- Possuem serviço de internet (93.00%) OK
- Participam em redes sociais (74.03%) NOTOK
- Possuem predominância em catalogo online (64.87%) NOTOK
- Possuem predominâncai em ter outras bases (69.29%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (98,6) NOTOK
- Maioria dos anos da IES são Faculdade (99.89%) NOTOK

- Pouca quantidade de técnicos na média (12.87) NOTOK
- Quantidade de alunos média de alunos abaixo (153.83) NOTOK
- Corpo docente muito pouco na média (14.20) NOTOK 
- Pouca quantidade de cursos (2.30) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOTOK

Insights falsos:
- A análise não detalha quais as características e nem a relevancia delas
- é uma análsie superficial, como se houvesse criado um gráfico de dispersão para analisar. 



In [235]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'segundo', 2, 17, 5, 12, 3, 0]

### Third Cluster

In [236]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'terceiro'))

**Análise do Cluster**

Ao receber o CSV com as instâncias de um determinado cluster, meu primeiro passo é entender o objetivo da análise e os fatores que podem influenciar a criação desse agrupamento.

**Perguntas Iniciais**

Para começar, eu me pergunto:

* Qual é o objetivo do cluster? (por exemplo: segmentação de mercado, segmentação de clientes)
* Quais são as características relevantes para o cluster?
* Há alguma informação adicional disponível que possa ajudar na análise?

**Análise das Características**

Depois de identificar as características relevantes, eu procedo com a análise estatística das mesmas. Aqui estão algumas perguntas que eu faria:

* Qual é a distribuição dos valores das características? (por exemplo: normal, fatial, uniforme)
* Há variação significativa nas características ao longo do dataset?
* Existem padrões ou tendências nas características?

**Análise das Conexões entre Características**

Em seguida, eu analiso as conexões entre as características, procura

Conclusões:

- Predominância das regiões Sudeste e Nordeste e Sul (47.17%, 19.17%, 17.48%) NOTOK
- Predominãncia das IES não estarem numa capital (62.70%) NOTOK
- Instituições sem acesso ao portal Capes (83.09%) NOTOK
- Possuem predominância em não possuir repositorio institucional (74.68%) NOTOK
- Possuem predominância em serviço de internet (79.13%) NOTOK
- Predominância em participar em redes sociais (62.62%) NOTOK
- Possuem predominância em ter outras bases (75.57%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (97,73)  NOTOK
- Maioria dos anos da IES são Faculdade (99.92%) NOTOK

- Quantidade baixa de média de técnicos (10.78) NOTOK
- Quantidade de alunos média de alunos baixo (185.81) NOTOK
- Corpo docente muito baixo na média (16.72) NOTOK
- Pouca quantidade de cursos (2.03) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOTOK

Insights falsos:
- A análise descreve etapas para um processo de análise do cluster, mas não análisa o cluster e o contexto da base de dados. 

In [237]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'terceiro', 0, 15, 2, 13, 3, 1]

In [238]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'terceiro'))

**Análise do Cluster**

Ao analisar o dataset fornecido, podemos identificar alguns fatores que podem ter contribuído para a criação do cluster.

**Características isoladas**

A primeira coisa que chamou a atenção foi a presença de características isoladas que não estão conjugadas com outras características. Por exemplo, a característica "Idade" está presente em todas as instâncias do cluster, mas não está relacionada com outras características como "Salário", "Nível de Educação" ou "Localização".

**Características conjugadas**

Outro fator que chamou a atenção foi a presença de características conjugadas. Por exemplo, as instâncias do cluster têm valores próximos entre si para as características "Salário" e "Nível de Educação". Isso pode indicar que essas características estão relacionadas e podem ter contribuído para o agrupamento.

**Distribuição das características**

A análise da distribuição das características também é importante. Por exemplo, a característica "Salário" tem uma

Conclusões:

- Predominância das regiões Sudeste e Nordeste e Sul (47.17%, 19.17%, 17.48%) NOTOK
- Predominãncia das IES não estarem numa capital (62.70%) NOTOK
- Instituições sem acesso ao portal Capes (83.09%) NOTOK
- Possuem predominância em não possuir repositorio institucional (74.68%) NOTOK
- Possuem predominância em serviço de internet (79.13%) NOTOK
- Predominância em participar em redes sociais (62.62%) NOTOK
- Possuem predominância em ter outras bases (75.57%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (97,73) OK
- Maioria dos anos da IES são Faculdade (99.92%) NOTOK

- Quantidade baixa de média de técnicos (10.78) NOTOK
- Quantidade de alunos média de alunos baixo (185.81	) NOTOK
- Corpo docente muito baixo na média (16.72) NOTOK
- Pouca quantidade de cursos (2.03) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOTOK

Insights:
- A análise reconhece que se trata de informações de Nível superior, mas não apresneta detalhes.
- Também cria variaveisfora de contexto, como idade.  

In [239]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'terceiro', 1, 15, 5, 10, 2, 4]

In [240]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'terceiro'))

Entendo o desafio! Vou te guiar por uma análise detalhada dos dados estatísticos para entender os fatores que levaram à criação do cluster.

**Passo 1: Obter visão geral**

Antes de começar, é importante ter uma visão geral do dataset fornecido. Nesse caso, o CSV contém as instâncias de um determinado cluster. Vamos observar os dados e entender se há algum padrão ou tendência aparente.

**Passo 2: Analisar características individuais**

Vamos analisar cada característica individualmente para entender se há algum padrão ou tendência significativa. Isso inclui calcular medidas de tendência central (média, mediana, moda) e dispersão (desvio padrão, variança).

**Observação:** Ao analisar as características individuais, observamos que a característica "features" tem uma distribuição simétrica com um desvio padrão de 0,5. A característica "target" também tem uma distribuição simétrica com um desvio padrão de 0,2.

**Passo 3: Analisar características conjuntas**

Agora, vamos analisar as car

Conclusões:

- Predominância das regiões Sudeste e Nordeste e Sul (47.17%, 19.17%, 17.48%) NOTOK
- Predominãncia das IES não estarem numa capital (62.70%) NOTOK
- Instituições sem acesso ao portal Capes (83.09%) NOTOK
- Possuem predominância em não possuir repositorio institucional (74.68%) NOTOK
- Possuem predominância em serviço de internet (79.13%) NOTOK
- Predominância em participar em redes sociais (62.62%) NOTOK
- Possuem predominância em ter outras bases (75.57%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (97,73) NOTOK
- Maioria dos anos da IES são Faculdade (99.92%) NOTOK

- Quantidade baixa de média de técnicos (10.78) NOTOK
- Quantidade de alunos média de alunos baixo (185.81) NOTOK
- Corpo docente muito baixo na média (16.72) NOTOK
- Pouca quantidade de cursos (2.03) NOTOK
- Despesas e receitas média moderadas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixa NOTOK



Insights:
- A análise foca em detalhar o processo de criação de um cluster. 

In [241]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'terceiro', 2, 15, 7, 8, 1, 1]

### Forth Cluster

In [242]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'quarto'))

Vamos analisar o dado fornecido!

A primeira coisa que fiz é verificar se há alguma característica isolada que possa ser identificada como um "ponto" inicial do cluster. Isso significa que uma única instância de uma característica pode estar presente no conjunto de dados e que possa ter sido a origem da formação do agrupamento.

Ao analisar as características, observei que existem algumas características que são muito consistentes entre si. Por exemplo:

* A característica "idade" tem valores muito próximos entre si, com uma média de 30 anos e uma amplitude padrão de 5 anos.
* A característica "salário" também tem valores próximos, com uma média de R$ 5.000,00 e uma amplitude padrão de R$ 1.000,00.

Essas características consistentes sugerem que o cluster pode estar relacionado a alguma característica mais amplamente presente no conjunto de dados.

A próxima coisa que fiz é verificar se há alguma característica que esteja conjuntamente com essas características consistentes. Isso signi

Conclusões:

- Predominância dos anos 2018 e 2019 (60,08%) NOTOK
- Predominância das regiões Sudeste, Sul e Nordeste (82,92%) NOTOK
- Uma incidência das IES não estarem numa capital (68.85%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes ( 64.73%) NOTOK
- Possuem predominância em busca integrada (77.29%) NOTOK
- Possuem serviço de internet (97.45%) NOTOK
- Predominância em participar em redes sociais (71.20%) NOTOK
- Possuem predominância em ter catalogo online (79.45%) NOTOK
- Possuem predominância em ter outras bases (73.43%) NOTOK
- Maioria das iinstituições são privada (com e sem fins lucrativos) (98,04) NOTOK
- Maioria dos anos da IES são Faculdade (99.15%) NOTOK

- Quantidade de técnicos na média baixo (33.68	) NOTOK
- Quantidade de alunos média de alunos  moderado (932.89) NOTOK
- Corpo docente muito na média (45.12) NOTOK
- Pouca quantidade média de cursos (6.21) NOTOK
- Despesas e receitas média acima do moderado NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights falsos:
- Destaca a necessidade de análises aprofundadas
- Descreve as etapas, mas não realiza a análise das variaveis. 



In [243]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'quarto', 0, 17, 3, 14, 3, 2]

In [244]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'quarto'))

Para iniciar essa análise, é fundamental entender o conceito de cluster e suas implicações estatísticas. Um cluster é um agrupamento de dados próximos uns aos outros na estrutura ou no espaço. Em outras palavras, os dados dentro de um cluster têm características semelhantes.

**Pertencimento dos Dados ao Cluster**

O primeiro passo que tomaria é analisar a distribuição dos dados para entender como eles estão agrupados. Será necessário visualizar as distâncias entre cada instância do dado e os pontos de referência. Uma das formas mais comuns de representar essas relações são utilizando a técnica de Agulha-Flecha, que permite visualizar uma distribuição dos dados em um plano 2D.

**Características Conjugadas**

A próxima etapa seria identificar características conjugadas dentro do cluster, ou seja, variáveis estatísticas que estão relacionadas entre si. Isso pode ser feito por meio da análise de correlação e co-variância.

Por exemplo, se estamos lidando com dados de clientes, uma caract

Conclusões:

- Predominância dos anos 2018 e 2019 (60,08%) NOTOK
- Predominância das regiões Sudeste, Sul e Nordeste (82,92%) NOTOK
- Uma incidência das IES não estarem numa capital (68.85%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes ( 64.73%) NOTOK
- Possuem predominância em busca integrada (77.29%) NOTOK 
- Possuem serviço de internet (97.45%) NOTOK
- Predominância em participar em redes sociais (71.20%) NOTOK
- Possuem predominância em ter catalogo online (79.45%) NOTOK
- Possuem predominância em ter outras bases (73.43%) NOTOK
- Maioria das iinstituições são privada (com e sem fins lucrativos) (98,04) NOTOK
- Maioria dos anos da IES são Faculdade (99.15%) NOTOK

- Quantidade de técnicos na média baixo (33.68	) NOTOK
- Quantidade de alunos média de alunos  moderado (932.89) NOTOK
- Corpo docente muito na média (45.12) NOTOK
- Pouca quantidade média de cursos (6.21) NOTOK
- Despesas e receitas média acima do moderado NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights:
- A análise acaba sendo das etapas de análise do cluster,e não das variaveis.





In [245]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'quarto', 1, 17, 7, 10, 3, 0]

In [246]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'quarto'))

Vamos analisar o dado de forma estruturada e identificar os fatores que podem ter contribuído para a criação do cluster.

**Orientação**

A primeira coisa que notamos é que as instâncias do cluster apresentam uma distribuição simétrica nos valores, o que sugere que não há um padrão ou uma tendência dominante. Isso indica que o agrupamento pode ser considerado representativo da população em questão.

**Características**

Agora, vamos analisar as características individuais das instâncias do cluster. Embora não seja possível identificar uma única característica que "defina" o cluster, notamos que há alguns padrões interessantes:

*   **Variação nos valores de uma característica**: Uma característica específica apresenta uma variação mais alta do que a média, indicando uma distribuição de valores mais ampla. Isso pode sugerir que essa característica é crítica para o agrupamento.
*   **Conjunção entre características**: Algumas características estão conjugadas, ou seja, apresentam um relac

Conclusões:

- Predominância dos anos 2018 e 2019 (60,08%) NOTOK
- Predominância das regiões Sudeste, Sul e Nordeste (82,92%) NOTOK
- Uma incidência das IES não estarem numa capital (68.85%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes ( 64.73%) NOTOK
- Possuem predominância em busca integrada (77.29%) NOTOK
- Possuem serviço de internet (97.45%) NOTOK
- Predominância em participar em redes sociais (71.20%) NOTOK
- Possuem predominância em ter catalogo online (79.45%) NOTOK
- Possuem predominância em ter outras bases (73.43%) NOTOK
- Maioria das iinstituições são privada (com e sem fins lucrativos)  (98,04) OK
- Maioria dos anos da IES são Faculdade (99.15%) OK

- Quantidade de técnicos na média baixo (33.68	)  NOTOK
- Quantidade de alunos média de alunos  moderado (932.89) NOTOK
- Corpo docente muito na média (45.12) NOTOK
- Pouca quantidade média de cursos (6.21) NOTOK
- Despesas e receitas média acima do moderado NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights falsos:
- A análise que é fieta acaba sendo da distribuição dos dados e não das caracteristicas das variaveis. 


In [247]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'quarto', 2, 17, 4, 13, 3, 1]

### Firth Cluster

In [248]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'quinto'))

**Análise do Clustering**

Ao receber o CSV contendo as instâncias de um determinado cluster, é necessário primeiro entender a estrutura dos dados e identificar os fatores que podem estar relacionados à formação desse agrupamento.

**Pensamento Inicial**

A primeira pergunta que se faz é: "Qual é a característica principal que está ligando as instâncias do CSV?".

Uma possível abordagem é analisar a média e a variança das características para entender se há algum padrão ou tendência dominante. Além disso, é importante considerar a distribuição das características, como se elas seguem uma distribuição normal ou se há uma tendência de extremos.

**Análise dos Fatores**

Ao analisar os dados, é possível identificar alguns fatores que podem estar relacionados à formação do cluster:

*   **Característica 'Age'**: A idade das instâncias do CSV parece ser um fator importante. As instâncias mais velhas estão concentradas em um lado do gráfico, enquanto as mais jovens estão no outro.
*   **Cara

Conclusões:

- Predominância das regiões Sudeste e Nordeste (69,45%) NOTOK
- Uma incidência das IES não estarem numa capital (59.17%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes ( 63.55%) NOTOK
- Possuem predominância em busca integrada (77.90%) NOTOK
- Possuem serviço de internet (97.47%) NOTOK
- Predominância em participar em redes sociais (60.40%) NOTOK
- Possuem catalogo online (85.57%) NOTOK
- Possuem predominância em ter outras bases (78.28%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (96,85) OK
- Maioria dos anos da IES são Faculdade e centro universitário (66.16%, 32.23%) OK

- Quantidade de técnicos médio moderado (143.65) NOTOK
- Quantidade de alunos média de alunos acima do moderado (4,691.86) NOTOK
- Corpo docente muito acima da  média (151.18) NOTOK
- Moderada quantidade média de cursos (18.49) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights:
- Não realiza a análise das variaveis, apenas sugere etapas de analise e o contexto. 

In [249]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'quinto', 0, 16, 4, 12, 3, 0]

In [250]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'quinto'))

**Análise do Cluster**

Para realizar essa análise, é necessário entender o conceito de cluster e seus fatores que podem contribuir para sua formação. Um cluster é um agrupamento de pontos ou instâncias em um espaço de dados que compartilham características semelhantes.

**Características do Cluster**

Com base nos dados fornecidos, identificamos as seguintes características que podem ter contribuído para a criação desse cluster:

* **Densidade**: O cluster apresenta uma densidade relativamente alta, o que sugere que as instâncias estão próximas entre si em termos de valores.
* **Escala**: As instâncias do cluster variam em escala, com alguns valores sendo muito pequenos ou grandes em comparação com os outros.
* **Simetria**: O cluster apresenta uma certa simetria ao redor da média, o que sugere que as distâncias entre as instâncias são regulares.

**Fatores que Contribuíram para a Formação do Cluster**

Com base nas características observadas, podemos inferir os seguintes fatores que 

Conclusões:

- Predominância das regiões Sudeste e Nordeste (69,45%) NOTOK
- Uma incidência das IES não estarem numa capital (59.17%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes ( 63.55%) NOTOK
- Possuem predominância em busca integrada (77.90%) NOTOK
- Possuem serviço de internet (97.47%) NOTOK
- Predominância em participar em redes sociais (60.40%) NOTOK
- Possuem catalogo online (85.57%) NOTOK
- Possuem predominância em ter outras bases (78.28%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (96,85) NOTOK
- Maioria dos anos da IES são Faculdade e centro universitário (66.16%, 32.23%) NOTOK

- Quantidade de técnicos médio moderado (143.65) NOTOK
- Quantidade de alunos média de alunos acima do moderado (4,691.86) NOTOK
- Corpo docente muito acima da  média (151.18) NOTOK
- Moderada quantidade média de cursos (18.49) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights :
- Apenas detalha o processo de criação do cluster, mas não análisa a relevancia das variaveis. 

In [251]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'quinto', 1, 16, 5, 11, 3, 1]

In [252]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'quinto'))

Para realizar essa análise, vou seguir um processo sistemático para entender as características do cluster fornecido e identificar os fatores que levaram à sua criação.

**Passo 1: Visualização Inicial**

Primeiramente, é importante visualizar o conjunto de dados de forma geral, usando uma representação gráfica como um grafico com todas as características por lado. Isso nos dá uma ideia da distribuição dos dados e identifica possíveis agrupamentos.

**Passo 2: Identificação das Características Principais**

A seguir, identificarei as características que têm a maior influência sobre o cluster. Isso pode ser feito selecionando as características com a maior variação de valores entre os instâncias do cluster ou observando quais são as características mais significativas em termos de distribuição.

**Passo 3: Análise Estatística**

Realizarei análises estatísticas para entender melhor as características que levaram à formação do cluster. Isso pode incluir:

*   **Média**: Verificar a média

Conclusões:

- Predominância das regiões Sudeste e Nordeste (69,45%) NOTOK
- Uma incidência das IES não estarem numa capital (59.17%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes ( 63.55%) NOTOK
- Possuem predominância em busca integrada (77.90%) NOTOK
- Possuem serviço de internet (97.47%) NOTOK 
- Predominância em participar em redes sociais (60.40%) NOTOK
- Possuem catalogo online (85.57%) NOTOK
- Possuem predominância em ter outras bases (78.28%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (96,85) NOTOK
- Maioria dos anos da IES são Faculdade e centro universitário (66.16%, 32.23%) NOTOK

- Quantidade de técnicos médio moderado (143.65) NOTOK
- Quantidade de alunos média de alunos acima do moderado (4,691.86) NOTOK
- Corpo docente muito acima da  média (151.18) NOTOK
- Moderada quantidade média de cursos (18.49) NOTOK
- Despesas e receitas média altas NOTOK 
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights falsos:
- A analise que é feita é sobre a distribuição do cluster. 

In [253]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'quinto', 2, 16, 6, 10, 1, 0]

### Sixth Cluster

In [254]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'sexto'))

Vamos a análise!

**Passo 1: Entendendo o Dados**

Antes de começar a analisar os dados, é importante entender o contexto e a natureza dos dados que foram fornecidos. No entanto, não tenho acesso ao CSV atualmente. Suponha que tenhamos um conjunto de dados com várias características ou atributos que definem as instâncias do cluster.

**Passo 2: Identificando Características Importantes**

Uma das primeiras etapas é identificar quais são as características mais importantes que definem o cluster. Isso pode ser feito analisando a distribuição dos dados, medindo a variância e a dispersão das instâncias.

Por exemplo, se temos um conjunto de dados com características como idade, renda, educação e localização, podemos começar a analisar como essas características estão relacionadas entre si e ao cluster.

**Passo 3: Identificando Fatores que Levaram à Criação do Cluster**

Agora é hora de examinar os fatores que podem ter contribuído para a criação desse cluster. Isso pode incluir:

*   Cara

Conclusões:

- Predominância dos anos 2016 e 2017 (35.63%, 28.14%) NOTOK
- Predominância das regiões Sudeste e Nordeste (43.65%, 20.80%)  NOTOK
- Predominância das IES não estarem numa capital (69.21%)  NOTOK
- Predominância das Instituições sem acesso ao portal Capes (73.15%)  NOTOK
- Predominância das Instituições sem repositorio institucional (68.00%)  NOTOK
- Possuem predominância em busca integrada (65.43%) NOTOK
- Possuem serviço de internet (90.62%) NOTOK
- Predominância em participar em redes sociais (65.28%)  NOTOK
- Possuem catalogo online (72.39%)  NOTOK
- Possuem predominância em ter outras bases (62.25%)  NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (90,01%)  NOTOK
- Maioria dos anos da IES são Faculdade (95.39%)  NOTOK

- Quantidade de técnicos médio baixo (46.05	) NOTOK
- Quantidade de alunos média de alunos  moderado (1,202.21) NOTOK
- Corpo docente moderado na  média (60.92) NOTOK
- Moderada quantidade baixa de cursos (16.76) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixo NOTOK

Insights falsos:
- Analisa a distibuição do cluster, mas não leva em consideraçãpo o contexto. 


In [255]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'sexto', 0, 18, 9, 9, 1, 0]

In [256]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'sexto'))

Entendo! Vamos analisar o conjunto de dados fornecido para entender melhor os fatores que levaram à criação do cluster.

**Análise dos Dados**

O primeiro passo é entender a natureza dos dados e a distribuição das características dentro do cluster. Nós vamos considerar as seguintes características:

* **Peso (Weight)**: A característica de peso parece ser a mais importante, pois a maioria das instâncias tem valores próximos entre 20 e 50 kg.
* **Altura (Height)**: A característica de altura também é relevante, com valores que variam entre 1.40m e 1.80m.
* **Idade (Age)**: A característica de idade apresenta uma distribuição mais uniforme, com valores que variam entre 20 e 60 anos.

**Análise de Conjugacões entre Características**

Para entender como as características estão relacionadas, vamos analisar as conjugacções entre elas. Vamos considerar as seguintes conjugacões:

* **Peso x Altura**: As instâncias com pesos próximos entre 20 e 30 kg tendem a ter alturas mais baixas (1.40m - 1

Conclusões:

- Predominância dos anos 2016 e 2017 (35.63%, 28.14%) NOTOK
- Predominância das regiões Sudeste e Nordeste (43.65%, 20.80%)  NOTOK
- Predominância das IES não estarem numa capital (69.21%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes (73.15%) NOTOK
- Predominância das Instituições sem repositorio institucional (68.00%) NOTOK
- Possuem predominância em busca integrada (65.43%) NOTOK
- Possuem serviço de internet (90.62%)  NOTOK
- Predominância em participar em redes sociais (65.28%) NOTOK
- Possuem catalogo online (72.39%) NOTOK
- Possuem predominância em ter outras bases (62.25%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (90,01%)  NOTOK
- Maioria dos anos da IES são Faculdade (95.39%)  NOTOK

- Quantidade de técnicos médio baixo (46.05	) NOTOK
- Quantidade de alunos média de alunos  moderado (1,202.21) NOTOK
- Corpo docente moderado na  média (60.92) NOTOK
- Moderada quantidade baixa de cursos (16.76) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relev ncia baixo NOTOK

Insights falsos:
- A análise cria variaveis que não estão no contexto do projeto. 

In [257]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'sexto', 1, 18, 4, 14, 1, 0]

In [258]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'sexto'))

Entendi! Vou agora proceder com o análise dos dados fornecidos, considerando os agrupamentos fornecidos como entrada.

**Análise Inicial**

Antes de iniciar a análise detalhada, é importante lembrar que o objetivo é entender os fatores que levaram à criação do cluster. Isso significa que precisamos identificar as características mais importantes e as relações entre elas.

**Estrutura dos Dados**

O CSV fornecido contém as instâncias de um determinado cluster, com várias características estatísticas associadas a cada instância. As características incluem:

* Media da idade (MED_IDADE)
* Padrão deviada da idade (STD_IDADE)
* Variância da idade (VAR_IDADE)
* Quantidade de registros por gênero (QNT_GÊNERO)

**Análise das Características**

Agora, vamos analisar cada característica individualmente:

1. **Media da Idade**: A média da idade é um valor central para a característica. Em geral, um valor alto indica uma idade mais avançada.
2. **Padrão Deviada da Idade**: O padrão deviada é uma m

Conclusões:

- Predominância dos anos 2016 e 2017 (35.63%, 28.14%) NOTOK
- Predominância das regiões Sudeste e Nordeste (43.65%, 20.80%)  NOTOK
- Predominância das IES não estarem numa capital (69.21%) NOTOK
- Predominância das Instituições sem acesso ao portal Capes (73.15%) NOTOK
- Predominância das Instituições sem repositorio institucional (68.00%) NOTOK
- Possuem predominância em busca integrada (65.43%) NOTOK
- Possuem serviço de internet (90.62%) NOTOK 
- Predominância em participar em redes sociais (65.28%) NOTOK
- Possuem catalogo online (72.39%) NOTOK
- Possuem predominância em ter outras bases (62.25%) NOTOK
- Maioria das instituições são privada (com e sem fins lucrativos) (90,01%)  OK
- Maioria dos anos da IES são Faculdade (95.39%) OK
 
- Quantidade de técnicos médio baixo (46.05	) NOTOK
- Quantidade de alunos média de alunos  moderado (1,202.21) NOTOK
- Corpo docente moderado na  média (60.92) NOTOK
- Moderada quantidade baixa de cursos (16.76) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância baixo NOTOK

Insights falsos:
- Suposições incondizentes com a realidade;
- A análsie não é feita para as avariaveis e sim sobre uma suposta distribuição do cluster. 


In [259]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'sexto', 2, 18, 3, 15, 0, 0]

### Seventh Cluster

In [260]:
print(show_result_kmeans_prompt(kmeans_fase2, 0, 'setimo'))

**Análise do Cluster**

Ao receber um conjunto de dados em formato CSV relacionado a um determinado cluster, é fundamental realizar uma análise cuidadosa para entender os fatores que levaram à sua criação. Neste caso, vamos proceder com uma abordagem estatística para identificar as características-chave e as relações entre elas.

**Características Principais**

A primeira etapa é analisar as características principais do conjunto de dados. Isso inclui:

1. **Descrição Estatística**: Verificar se há padrões ou tendências na distribuição das variáveis numéricas (e.g., média, mediana, moda) e em categorias (e.g., porcentagem de valores).

2. **Análise de Correlação**: Avaliar as relações entre as características. Isso pode ajudar a identificar grupos ou subgrupos dentro do conjunto de dados.

3. **Gráficos de Distribuição**: Usar gráficos de distribuição (como histogramas, boxplots) para visualizar a forma como as variáveis estão espalhadas e se há algum desequilíbrio na representação dos

Conclusões:

- Predominância das regiões Sudeste e Sul (44.90%, 24.32%) NOTOK
- Predominância das Instituições com acesso ao portal Capes (87.76%) NOTOK
- Predominância das Instituições com repositorio institucional (70.41%)  NOTOK
- Possuem predominância em busca integrada (67.18%) NOTOK
- Possuem serviço de internet (98.81%) NOTOK
- Predominância em participar em redes sociais (71.94%)  NOTOK
- Possuem catalogo online (92.69%)  NOTOK
- Possuem predominância em ter outras bases (91.16%)  NOTOK
- Maioria das instituições são privada sem fins, federal e com fins lurativos (46.26%, 25.68%, 19.90%)  NOTOK
- Maioria dos anos da IES são Universidade e centro universitario (69.73%, 22.28%)  NOTOK

- Quantidade de técnicos médio alto (1,170.62	) NOTOK
- Quantidade de alunos média de alunos alto (27,666.31) NOTOK
- Corpo docente alto na  média (970.69) NOTOK
- altga quantidade baixa de cursos (73.91) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights falsos:
- Detalha uma etapa de análise dos clusters, mas não analisa o cluster. 

In [261]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'setimo', 0, 16, 7, 9, 1, 0]

In [262]:
print(show_result_kmeans_prompt(kmeans_fase2, 0.5, 'setimo'))

Vamos entender melhor o problema e fazer a análise dos dados fornecidos.

**Entendendo os Dados**

Antes de começarmos a analisar os dados, é importante saber que um cluster é uma coleção de instâncias que compartilham características semelhantes. Nesse caso, vamos assumir que o CSV fornece as instâncias do cluster com suas características.

**Pensamento Inicial**

O primeiro passo é entender como os agrupamentos foram criados. Perguntas que podemos fazer para entender melhor os dados:

* Quais são as características dos dados?
* Qual é a relação entre essas características e o agrupamento?
* Há algum padrão ou tendência nos dados?

**Análise dos Dados**

Vamos começar a analisar os dados. Vou considerar cada característica separadamente e tentar encontrar relações com as outras características.

* Característica 1: Valor do atributo A
 + Parece que há uma distribuição de valores simétrica ao redor da média, com alguns valores muito altos e baixos.
 + Pode ser interessante investigar s

Conclusões:

- Predominância das regiões Sudeste e Sul (44.90%, 24.32%) NOTOK
- Predominância das Instituições com acesso ao portal Capes (87.76%) NOTOK
- Predominância das Instituições com repositorio institucional (70.41%) NOTOK
- Possuem predominância em busca integrada (67.18%)  NOTOK
- Possuem serviço de internet (98.81%)  NOTOK
- Predominância em participar em redes sociais (71.94%)  NOTOK
- Possuem catalogo online (92.69%) NOTOK
- Possuem predominância em ter outras bases (91.16%)
- Maioria das instituições são privada sem fins, federal e com fins lurativos (46.26%, 25.68%, 19.90%)  NOTOK
- Maioria dos anos da IES são Universidade e centro universitario (69.73%, 22.28%)  NOTOK

- Quantidade de técnicos médio alto (1,170.62	) NOTOK
- Quantidade de alunos média de alunos alto (27,666.31) NOTOK
- Corpo docente alto na  média (970.69) NOTOK
- altga quantidade baixa de cursos (73.91) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights falsos:
- A análise feita é da distribuição do cluster, mas não das características do cluster. 

In [263]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'setimo', 1, 16, 7, 9, 1, 0]

In [264]:
print(show_result_kmeans_prompt(kmeans_fase2, 1.0, 'setimo'))

**Análise do Clustering**

Recebemos um conjunto de dados em CSV que representa instâncias de um determinado cluster. O objetivo é analisar os agrupamentos presentes no conjunto de dados e identificar os fatores que levaram à sua criação.

**Características do Dados**

Após analisar as informações disponíveis, podemos observar que o conjunto de dados é composto por 100 instâncias com 10 características cada uma. As características são:

1. **ID**: número único identificador para cada instância
2. **Ação**: tipo de ação realizada (compra ou venda)
3. **Preço**: preço do produto
4. **Quantidade**: quantidade de produtos vendidos
5. **Faturamento**: faturamento total gerado pela venda
6. **Tempo**: tempo médio de entrega
7. **Localização**: localização geográfica da loja
8. **Categoria**: categoria do produto (roupa, eletrônica, etc.)
9. **Tipo de Mercado**: tipo de mercado (retail ou online)
10. **Análise Fatorial**: resultado de uma análise fatorial realizada sobre as características an

Conclusões:

- Predominância das regiões Sudeste e Sul (44.90%, 24.32%) NOTOK
- Predominância das Instituições com acesso ao portal Capes (87.76%)  NOTOK
- Predominância das Instituições com repositorio institucional (70.41%)  NOTOK
- Possuem predominância em busca integrada (67.18%)  NOTOK
- Possuem serviço de internet (98.81%)  NOTOK
- Predominância em participar em redes sociais (71.94%)  NOTOK
- Possuem catalogo online (92.69%)  NOTOK
- Possuem predominância em ter outras bases (91.16%)  NOTOK
- Maioria das instituições são privada sem fins, federal e com fins lurativos (46.26%, 25.68%, 19.90%)  NOTOK
- Maioria dos anos da IES são Universidade e centro universitario (69.73%, 22.28%)  NOTOK

- Quantidade de técnicos médio alto (1,170.62	) NOTOK
- Quantidade de alunos média de alunos alto (27,666.31) NOTOK
- Corpo docente alto na  média (970.69) NOTOK
- altga quantidade baixa de cursos (73.91) NOTOK
- Despesas e receitas média altas NOTOK
- Quantidade de periodicos e livros eletronicos com relevância alto NOTOK

Insights:
- A análise cria outras variaveis fora do contexto do cluster. 

In [265]:
analise_kmeans.loc[len(analise_kmeans)] = ['Fase 2',  'setimo', 2, 16, 9, 7, 3, 0]